microservice_utils/pinecone/adapters.py

```diff
@@ -60,6 +60,12 @@ def query(
             for r in results
         ]
 
+    def delete(self, ids: typing.List[str]):
+        """Remove items from the Pinecone index."""
+        pinecone.init(api_key=self._api_key, environment=self._environment)
+        index = pinecone.Index(index_name=self._index_name)
```

Comment:
It'd be good to abstract these initiation lines into a protected method, because now we're using them in 3 instances.

```
    pinecone.init(api_key=self._api_key, environment=self._environment)
    index = pinecone.Index(index_name=self._index_name)
```